src/metabase/driver/settings.clj

```diff
@@ -177,3 +177,11 @@
   :getter     (fn []
                 ((requiring-resolve 'metabase.driver.util/available-drivers-info)))
   :doc        false)
+
+(defsetting mongodb-sync-field-limit
```

Comment:
Any chance other document-oriented databases have a notion of leaf field sync limits too? Wondering if this could be more generic